### PR TITLE
Latest changes

### PR DESCRIPTION
**Pull request by zharinovAlex**

### CODE REVIEW

File: src/main/java/tasker/tasker/controller/TaskController.java [hunk 1]
```diff
@@ -40,26 +38,26 @@ public TaskController(TaskService taskService) {
...
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
-    public void createTask(@Valid @RequestBody Task task) {
-        this.taskService.createTask(task);
+    public void createTask(@Valid @RequestBody TaskCreateDto dto) {
+        this.taskService.createTask(dto);
     }
 
-    @PatchMapping(value = "/{id}")
+    @PatchMapping(value = "/{taskId}")
```
Comment (ohrytsenko) [inline src/main/java/tasker/tasker/controller/TaskController.java]: to static constant

File: src/main/java/tasker/tasker/controller/UserController.java [hunk 2]
```diff
@@ -45,30 +42,35 @@ public UserController(
...
     @GetMapping(value = "/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity<UserPageDTO> getUser(@PathVariable Long id) {
+    public ResponseEntity<UserPageDto> getUser(@PathVariable Long id) {
         return ResponseEntity.ok().body(this.userService.convertToPageDto(id));
     }
 
     @GetMapping(value = "/{userId}/tasks", produces = MediaType.APPLICATION_JSON_VALUE)
     public @ResponseBody
-    Page<Task> getTasksForUser(
+    List<TaskListDto> getTasksForUser(
```
Comment (ohrytsenko) [inline src/main/java/tasker/tasker/controller/UserController.java]: inject pageable
pls refer to https://www.baeldung.com/rest-api-pagination-in-spring

File: src/main/java/tasker/tasker/controller/UserController.java [hunk 3]
```diff
@@ -45,30 +42,35 @@ public UserController(
...
+                .map(this.userService::convertToTaskListDto)
+                .collect(Collectors.toList());
     }
 
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
-    public void createUser(@Valid @RequestBody User user) {
-        this.userService.createUser(user);
+    public void createUser(@Valid @RequestBody UserCreateDto dto) {
+        this.userService.createUser(dto);
```
Comment (ohrytsenko) [inline src/main/java/tasker/tasker/controller/UserController.java]: remove dto from service layer

File: src/main/java/tasker/tasker/service/UserService.java [hunk 4]
```diff
@@ -86,18 +92,34 @@ public UserListDTO convertToListDto(User user) {
...
         dto.setEmail(user.getEmail());
         dto.setUpdatedAt(user.getUpdatedAt());
-        dto.setActive(user.isActive());
+        dto.setActive(user.getActive());
         dto.setTeam(user.getTeam());
 
         return dto;
     }
+
+    public TaskListDto convertToTaskListDto(Task task) {
```
Comment (ohrytsenko) [inline src/main/java/tasker/tasker/service/UserService.java]: consider using mapping libs

please refer to the following pages:
https://github.com/orika-mapper/orika
https://www.baeldung.com/rest-api-pagination-in-spring

File: src/main/java/tasker/tasker/controller/TaskController.java (hunk 1 above)
Comment (zharinovAlex) [reply src/main/java/tasker/tasker/controller/TaskController.java]: Added constant

File: src/main/java/tasker/tasker/controller/UserController.java (hunk 2 above)
Comment (zharinovAlex) [reply src/main/java/tasker/tasker/controller/UserController.java]: Changed to pageable lists

File: src/main/java/tasker/tasker/controller/UserController.java (hunk 3 above)
Comment (zharinovAlex) [reply src/main/java/tasker/tasker/controller/UserController.java]: Removed to controllers

**Review by ohrytsenko — APPROVED** [review]

**Pull request merged**